test/pummel/test-buffer-large-size.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const common = require('../common');
+
+// Buffer with size > INT32_MAX
+common.skipIf32Bits();
+
+// Test Buffer size larger than integer range
+const assert = require('assert');
+const {
+  SlowBuffer,
+} = require('buffer');
+const kStringMaxLength = require('buffer').constants.MAX_STRING_LENGTH;
```

Comment:
non-blocking nit: since this is a new test file, it would be helpful if the test could be structured a bit more using `node:test`. Up to you tho.

Reply:
@jasnell Resolved.